docs/data-importing.md

```diff
@@ -0,0 +1,124 @@
+# Whisper data import
+
+Metrictank comes with a set of tools to import Whisper data into the chunk format used by Metrictank. 
+The usage of the tools is documented on [this page](https://github.com/grafana/metrictank/blob/master/docs/tools.md)([reader](https://github.com/grafana/metrictank/blob/master/docs/tools.md#mt-whisper-importer-reader)/[writer](https://github.com/grafana/metrictank/blob/master/docs/tools.md#mt-whisper-importer-writer)), this document specifically focuses on how the import procedure of Whisper files works.
+
+## Import procedure
+
+### Determination of metric names
+
+The importer utility walks recursively through the whisper directory, looking for files which end with `.wsp`. If there is a name filter defined it additionally filters all the discovered files by the given pattern. It then generates the metric name by removing the prefix defined via the `-whisper-directory` parameter from the file's path, removing the string `.wsp` suffix, and replacing all the `/` characters with `.`.
+
+For example if a file is located at `/opt/graphite/storage/my/metric/name.wsp` and the value given to `-whisper-directory` is `/opt/graphite/storage`, then the resulting metric name would be `my.metric.name`.
+
+Additionally there is an optional parameter called `-name-prefix` which allows the caller to prefix all metric names with a given string, which needs to include the tailing `.` if it is wanted. In the above example, if the value of `-name-prefix` is `my.prefix.` then the resulting metric name would by `my.prefix.my.metric.name`.
+
+### Schema conversion
+
+The whisper importer reader requires the user to provide the path to the storage schemas used at the import destination (**not the one on the source graphite installation**), it then reads and parses those schemas. During the import it iterates over all the whisper files that need to be imported and processes each of them as following:
+
+* Reads all the headers and points
+* Generates the MetricData, interval gets set to the raw interval of the destination schema
+* Iterates over each archive in the destination schema (raw + all rollups), for each destination archive it uses the schema conversion logic described [in the next chapter](https://github.com/grafana/metrictank/blob/master/docs/data-importing.md#schema-conversion-logic) to generate the archive points
+* Generates chunks from the points which have been generated by the schema conversion logic
+* Chunks are then sent to the importer-writer, which writes them into the store used by the destination cluster
+
+Note that this conversion allows to convert from one storage-schema to another, but it does not support changes to the storage-aggregations.
```

Comment:
this could use more clarification.
If src aggregation says "avg", then we can't create "max". does that mean it will error? what will happen?
what happens if we want, say "avg" and "sum", or "avg", "sum" and "cnt"?

Reply:
none of the tools or functions in the code allow the user to specify what aggregation functions they want to be used. all the whisper-importer-reader currently does is to read the whisper header (which includes the aggregation function) and then it uses that aggregation function. 

if somebody would import data into a Metrictank which has a `storage-aggregations.conf` that specifies different aggregation functions than were used in the whisper files, then the resulting situation would be the same as if another Metrictank with different `storage-aggregations.conf` would have persisted its rollups in the store, the data just wouldn't be found.

Reply:
OK, this is super useful. can you please add this to the document?

Reply:
i have added it https://github.com/grafana/metrictank/pull/1594/commits/bff5e58c0f71e16c8f881615d200d96b35c12f1f